triage/summarize/cluster.go

```diff
@@ -0,0 +1,296 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/*
+Contains functions that cluster failures.
+*/
+
+package summarize
+
+import "time"
+
+/*
+clusterLocal clusters together the failures for each test. Failures come in grouped
+by the test they belong to. These groups are subdivided into clusters.
+
+Takes:
```

Comment:
I think it would be better here in this case since `clusterLocal()` is only tested through `TestSummarize`, not on its own, and `TestSummarize` is unfortunately very complex due to the complicated data structures.